### ISSUE TITLE
Poetry 1.4.2 update takes until the end of time

### ISSUE THREAD

**Issue by rjurney** [python-poetry/poetry#7794]
Originally described in [this gist](https://gist.github.com/rjurney/a55324ef75cf3c52154cbe4f1c2ca345).

- **Poetry version**: Poetry (version 1.4.1)
- **Python version**: Python 3.10.10
- **OS version and name**: Ubuntu 20.04.6 LTS
- **pyproject.toml**: https://gist.github.com/rjurney/a55324ef75cf3c52154cbe4f1c2ca345#file-pyproject-toml

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.

The FAQ talks about Poetry being slow - deal with it. This is not my experience just a few months ago and pytest is not a long list of packages to resolve... I think this is a bug. The `-vvv` output says nothing about inspecting packages.

- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

I have a new project. I was using `pyproject.toml` for parameters to black, isort, etc. using `requirements.txt`. I looked at a `pyproject.toml` from a previous project and edited it to get started using poetry for package management. The `poetry update` command does not work, it takes overnight and makes no progress. I reduced [my pyproject.toml](https://gist.github.com/rjurney/a55324ef75cf3c52154cbe4f1c2ca345#file-pyproject-toml) to a single dev dependency: `pytest = ">=7.3.0"` and it still will not update.

See: [`poetry.toml`](https://gist.github.com/rjurney/a55324ef75cf3c52154cbe4f1c2ca345#file-poetry-toml)
See: [`poetry update` command output](https://gist.github.com/rjurney/a55324ef75cf3c52154cbe4f1c2ca345#file-poetry_update_stalls-txt)

**Comment by dimbleby** [comment]
probably duplicate #7557, #1917, etc, please close if so

**Comment by rjurney** [comment]
@dimbleby Now I restarted with `poetry init` and I can't `poetry add` a single package: `black`. 

This was addressed by a suggestion in #7557 to set the `PYTHON_KEYRING_BACKEND` environment variable.

```bash
export PYTHON_KEYRING_BACKEND=keyring.backends.null.Keyring
```

See [my comment on #7557](https://github.com/python-poetry/poetry/issues/7557#issuecomment-1506095987)

**Comment by rjurney** [comment]
Closing as dupe of #7557.
